src/KOKKOS/pair_metatomic_kokkos.cpp

```diff
@@ -210,51 +238,55 @@ void PairMetatomicKokkos<DeviceType>::compute(int eflag, int vflag) {
         energy_tensor.backward(-torch::ones_like(energy_tensor));
 
         forces_tensor = this->system_adaptor->positions.grad();
-        virial_tensor = this->system_adaptor->strain.grad().to(torch::kCPU);
+        virial_tensor = this->system_adaptor->strain.grad();
     }
 
     {
         auto _ = MetatomicTimer("storing model output in LAMMPS data structures");
 
-        // move results to cpu for storing
-        auto energy_detached = energy_tensor.detach().to(torch::kCPU).to(torch::kFloat64);
-        auto energy_samples = energy_block->samples();
-
-        // store the energy returned by the model
-        torch::Tensor global_energy;
-        if (eflag_atom) {
-            assert(energy_samples->size() == 2);
-            assert(energy_samples->names()[0] == "system");
-            assert(energy_samples->names()[1] == "atom");
-
-            auto samples_values = energy_samples->values().to(torch::kCPU);
-            auto samples = samples_values.accessor<int32_t, 2>();
-
-            int64_t n_atoms = atom->nlocal + atom->nghost;
-            assert(samples_values.sizes() == mta_data->selected_atoms_values.sizes());
-
-            auto energies = energy_detached.accessor<double, 2>();
-            for (int64_t i=0; i<energy_samples->count(); i++) {
-                assert(samples[i][0] == 0);
-                // handle potentially out of order samples in
-                // the per-atom energy tensor
-                auto atom_i = samples[i][1];
-                assert(atom_i < n_atoms);
-                eatom[atom_i] += this->scale * energies[i][0];
+        // store the energy (if **not** in non-conservative mode)
+        if (!mta_data->non_conservative) {
+            // move results to cpu for storing
+            auto energy_detached = energy_tensor.detach().to(torch::kCPU).to(torch::kFloat64);
+
+            if (eflag_atom) {
```

Comment:
I don't understand this logic either. What if the model is non-conservative and has energies?